Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewCell.cs

```diff
@@ -0,0 +1,1272 @@
+        public AccessibleObject AccessibilityObject
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<AccessibleObject>() != null);
+                return default(AccessibleObject);
+            }
+        }
+
+        /// <summary>
+        /// Gets the column index for this cell.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The index of the column that contains the cell; -1 if the cell is not contained within a column.
+        /// </returns>
+        public int ColumnIndex
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<int>() >= -1);
+                return default(int);
+            }
+        }
+
+        // <summary>
+        // Gets the bounding rectangle that encloses the cell's content area.
+        // </summary>
+        // 
+        // <returns>
+        // The <see cref="T:System.Drawing.Rectangle"/> that bounds the cell's contents.
+        // </returns>
+        // <exception cref="T:System.ArgumentOutOfRangeException">The row containing the cell is a shared row.-or-The cell is a column header cell.</exception><exception cref="T:System.InvalidOperationException">The value of the <see cref="P:System.Windows.Forms.DataGridViewCell.ColumnIndex"/> property is less than 0, indicating that the cell is a row header cell.</exception><PermissionSet><IPermission class="System.Security.Permissions.EnvironmentPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.SecurityPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Flags="UnmanagedCode, ControlEvidence"/><IPermission class="System.Diagnostics.PerformanceCounterPermission, System, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/></PermissionSet>
+        // public Rectangle ContentBounds {get;}
+        
+        // <summary>
+        // Gets or sets the shortcut menu associated with the cell.
+        // </summary>
+        // 
+        // <returns>
+        // The <see cref="T:System.Windows.Forms.ContextMenuStrip"/> associated with the cell.
+        // </returns>
+        // public virtual ContextMenuStrip ContextMenuStrip {get; set;}
+        
+        /// <summary>
+        /// Gets the default value for a cell in the row for new records.
+        /// </summary>
+        /// 
+        /// <returns>
+        // An <see cref="T:System.Object"/> representing the default value.
+        /// </returns>
+        public virtual object DefaultNewRowValue
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<object>() == null);
```

Comment:
Although this holds for the default implementation, subclasses can return other values, so this shouldn't be a postcondition.